javascript/react-js/advanced-concepts.md

```diff
@@ -36,15 +36,19 @@ If you start writing larger applications you will, with certainty, encounter the
 
 You might have already heard about Redux. [Redux](https://react-redux.js.org) is the most popular state management system out there. It is not a part of React, but the two can be very easily combined. Together, they make up an extremely powerful duo. The purpose is to store your application's state in a single place, commonly called a "store". You then dispatch actions to the store, where a reducer will handle the state changes. The primary benefit of using a state management library is to prevent having to pass props through multiple levels of the component tree. A state management library is often only recommended for larger applications.
 
-#### <span id="higher-order-components">4. Higher-order Components</span>
+#### <span id="context-api">4. Context API</span>
+
+As your application becomes larger and you have multiple components to ensure reusability, you might find yourself passing props through a a lot of components in the middle or you might have a lot of components that require the same props. This pattern is known as `prop drilling` and to help avoid this, React provides the Context API that lets a parent component provide data to all the components in its tree without having to pass props. Imagine you have implemented an optional dark theme in your website and quite a few of your components require this theme data to correctly style itself - having a `context` of your theme will allow all children components have access to this data. The [Context API beta docs](https://beta.reactjs.org/learn/passing-data-deeply-with-context) is a great resource to further explore and practice your learning.
```

Comment:
I have some wording suggestions here. I've also pulled the React Beta Docs links to a new paragraph and changed how it's worded as well. 

```suggestion
As your application becomes larger and you have multiple components to ensure reusability, you might find yourself passing props through a lot of components in the middle or you might have a lot of components that require the same props. This pattern is known as `prop drilling`. To help avoid this, React provides the Context API that lets a parent component provide data to all the components in its tree without having to pass props. Imagine that you have implemented an optional dark theme in your website and quite a few of your components require this theme data to correctly style itself - having a `context` of your theme will allow all children components to have access to this data.

The [Passing Data Deeply with Context webpage](https://beta.reactjs.org/learn/passing-data-deeply-with-context) on React Docs Beta does an excellent job of explaining on how to use the Context API. Furthermore, if you wish to learn more advanced concepts in React, [React Docs Beta](https://beta.reactjs.org/) is a great resource to dig into.
```

Reply:
Ah damn it, shouldn't have missed the double a in the sentence.. Resolved all the suggestions in the pushed commit - they've all made the section flow much nicely!